dashboard/app/models/user.rb

```diff
@@ -926,15 +948,47 @@ def self.send_reset_password_instructions(attributes={})
       return user
     end
 
-    user = find_by_email_or_hashed_email(attributes[:email]) || User.new(email: attributes[:email])
-    if user && user.persisted?
```

Comment:
the `persisted?` check doesn't seem to have survived this PR- will this change in behavior affect any particular edge-cases you can think of?